packages/neptune-extension/src/common/components/checkout-modal/CheckoutModal.tsx

```diff
@@ -16,35 +36,85 @@ const CheckoutModal: React.FC<CheckoutModalProps> = ({
   platformNotebook,
   onClose,
 }) => {
+  const [ loading, setLoading ] = React.useState(false);
 
-  /*
-   * TODO:
-   * 1. Implement form fields and checkpoint selection.
-   * 2. Implement downloading checkpoint content.
-   *
-   * For the moment we open current content in new window.
-   */
+  const { notebook } = useSelector(getNotebookState)
+  const initialProjectId = () => notebook ? notebook.projectId : getDefaultProjectId();
+  const initialNotebookId = notebook ? notebook.id : undefined;
+
+  const [ projectId, projectInputProps, projectMetaProps ] = useSelectInputValue(
+    initialProjectId,
+    () => fetchProjectOptions('readable'),
+    []
+  );
+
+  const [ notebookId, notebookInputProps, notebookMetaProps ] = useSelectInputValue(
+    initialNotebookId,
+    () => fetchNotebookOptions(projectId),
+    [projectId]
+  );
+
+  const [ checkpointId, checkpointInputProps, checkpointMetaProps ] = useSelectInputValue(
+    undefined,
+    () => fetchCheckpointOptions(notebookId),
+    [notebookId]
+  );
 
   async function handleSubmit() {
-    const content = await platformNotebook.getContent();
+    if (checkpointId === undefined) {
```

Comment:
We should add projectId === undefined in here so we dont have to cast it later on